lib/filesystem.nix

```diff
@@ -54,4 +54,44 @@
       dir + "/${name}"
   ) (builtins.readDir dir));
 
+  # findSuffix: String -> Path -> [ Path ]
+  #
+  # Given a suffix and directory, return a filtered list containing the full
+  # path to all files matching the given suffix recursively.
+  #
+  # Example:
+  #   findSuffix ".h" "${glibc.dev}/include/protocols"
```

Comment:
This is a bad examples because it encourages IFD.
IFD isn't bad when it's the only option, but I've seen a couple of cases where people use it for things that are easily achieved without IFD in the depending derivation.

Could you change this to a fictional local source, like `findSuffix ".h" ./src`?

Reply:
Alright, I've done that. I've also shortened the list to a simple `foo.h` example since this is now more simple and generic and not encouraging IFD.